docs/cql-extensions.md

```diff
@@ -41,4 +41,38 @@ The option can be specified at `CREATE TABLE` or `ALTER TABLE` queries in the sa
 way as other options by using `WITH` clause:
 
     CREATE TABLE tbl ...
-    WITH paxos_grace_seconds=1234
\ No newline at end of file
+    WITH paxos_grace_seconds=1234
+
+## ALTER SESSION
+
+The `ALTER SESSION` statement can be used to configure session-specific settings,
+e.g. custom latency limits.
+
+Examples:
+```cql
+ALTER SESSION
+	SET latency_limit_for_reads = 50ms
+	AND latency_limit_for_writes = 20ms
+```
+```cql
+ALTER SESSION
+	DELETE latency_limit_for_reads
+```
+
```

Comment:
(Focusing review here)

Naming bikeshed: `read_timeout` and `write_timeout`. We don't actually limit the latency, we time out,
Proposal: add `timeout` (or `latency_limit`) that takes effect if for_read/for_write isn't specified. This way you can set a general latency limit with one value.

We should also have a way to specify that a session is not for realtime ops, that is, it's okay to send empty pages as keepalives instead of abandoning a slow query.

Reply:
Yeah, I even started calling these variables "timeouts" in docs and commit messages, since that's what they actually are. I'll rename them all to timeouts.

Reply:
I also considered adding `timeout` or `default_timeout`, but right now `read_timeout` adds a single branch to the select path, and `write_timeout` adds a single branch to the modification path. Having a default per-session parameter would add yet another branch, while not giving any huge benefit to the users - I think they are equally fine with setting `session.set_read_timeout(50ms).set_write_timeout(50ms)` as they would be with `session.set_timeout(50ms)`. Perhaps this idea can be revisited if we later expand the parameters to have tens of different timeouts, in which case it will definitely be helpful to add a default. It can be done later as a follow-up with very little effort, since the whole machinery would already be in place.